### ISSUE TITLE
maim capture the selection rectangle

### ISSUE THREAD

**Issue by vincentbernat** [naelstrof/maim#257]
Hey!

I am using:

```shell
maim -u -s -b 5 -l -c 0.3,0.4,0.6,0.4 \
            | xclip -selection clipboard -t image/png
```

Often, the selection rectangle is part of the screenshot.

```
 13:18 ❱ compton --version
1
 13:18 ❱ maim --version
v5.7.4
```

**Comment by foxpy** [comment]
Probably a good idea to leave more information about your config (picom config, window manager/desktop environment, etc)

**Comment by vincentbernat** [comment]
- [picom configuration](https://github.com/vincentbernat/i3wm-configuration/blob/master/dotfiles/picom.conf)
- i3 as a window manager

Also, it seems easier to reproduce by capturing an area than by capturing a window. I don't remember having the issue when capturing a window. Using `--noopengl` does not help. A very simple workaround is to use `-d 0.1`. I was thinking that maybe with a compositor, one has to wait for a vsync cycle before being sure the screen is up-to-date?

**Comment by foxpy** [comment]
Where did you install picom from? What is your distro? I want to reproduce your environment.

**Comment by 0Karakurt0** [comment]
Do you use any scaling on your screen?

**Comment by vincentbernat** [comment]
No scaling from xrandr point of view, but scaling by the toolkit (Xft/DPI, Gdk/WindowScalingFactor, stuff like that).

@foxpy, sorry, I did miss your questions. I am running Debian sid and picom comes from the picom Debian package.

**Comment by foxpy** [comment]
> I am running Debian sid and picom comes from the picom Debian package.

Uh, this leaves me with following two questions:
- how do I install Debian sid? I have never used Debian before and, seems like, I should install Debian testing from [here](https://cdimage.debian.org/cdimage/weekly-builds/) (?) and then convert it into sid via enabling unstable (sid) repositories?
- I am still not fully aware about your environment. What is your desktop environment? What is your GPU? What are your GPU drivers?

**Comment by vincentbernat** [comment]
You can install either testing or stable and update to unstable with `apt full-upgrade` after changing sources. But you can just test with testing, the differences with unstable are small. All my machines are with an AMD GPU. I use the radeon driver in the kernel and the modesetting driver in X. I don't remember if I had the issue with an Intel GPU, but I don't have the appropriate laptop anymore.

If you have trouble reproducing, maybe don't put too much effort on it. The `-d 0.1` workaround is fine for me. The issue may help people running into the same problem and they may provide additional inputs to see what's the common point among us could be.

**Comment by foxpy** [comment]
> If you have trouble reproducing, maybe don't put too much effort on it.

Right, currently I don't have an unused Radeon machine at my disposal, hence it would be really hard for me to reproduce the bug. Let's just leave this issue as it is.

**Comment by mangelozzi** [comment]
I have the same problem with rectangular selection screen shots, for example here, bottom right you can see a black dot of the cross hair (all images end up having that black dot):

![image](https://user-images.githubusercontent.com/19764314/232737274-c3800c38-96ae-455a-9c48-a67ec864fe20.png)

Also using i3wm, Ubuntu 22.04, installed via `sudo apt install -y maim`

**Comment by mosin311aa** [comment]
<img width="717" height="380" alt="Image" src="https://github.com/user-attachments/assets/a80a2b26-44f0-4118-9640-ac6650ce94d2" />

(Parts of this image have been blurred to not promote off topic projects)

I still experience this issue, dwm using picom, slop is excluded from animations or any other picom features
